packages/poaps/src/PoapsClient.ts

```diff
@@ -0,0 +1,89 @@
+import { CompassProvider } from '@poap-xyz/providers';
+import { POAP } from './domain/Poap';
+import { PaginatedPoapsResponse, PAGINATED_POAPS_QUERY } from './queries';
+import {
+  createBetweenFilter,
+  createFilter,
+  creatEqFilter,
+  createInFilter,
+  filterUndefinedProperties,
+} from './queries/utils';
+import { FetchPoapsInput } from './types';
+import { PaginatedResult, nextCursor } from '@poap-xyz/utils';
+
+/**
+ * Represents a client for working with POAP drops.
+ *
+ * @class DropsClient
+ */
+export class PoapsClient {
+  /**
+   * Creates a new DropsClient object.
+   *
+   * @constructor
+   * @param {CompassProvider} CompassProvider - The provider for the POAP compass API.
+   */
+  constructor(private CompassProvider: CompassProvider) {}
+
+  /**
+   * Fetches drops based on the specified input.
+   *
+   * @async
+   * @method
+   * @param {FetchPoapsInput} input - The input for fetching drops.
+   * @returns {Promise<PaginatedResult<Drop>>} A paginated result of drops.
+   */
+  async fetch(input: FetchPoapsInput): Promise<PaginatedResult<POAP>> {
+    const {
+      limit,
+      offset,
+      order,
+      chain,
+      collector_address,
+      id_order,
+      minted_date_from,
+      minted_date_to,
+      ids,
+      drop_id,
+    } = input;
+
+    const variables = {
+      limit,
+      offset,
+      orderBy: filterUndefinedProperties({
+        minted_on: order,
+        id: id_order,
+      }),
+      where: {
+        ...createFilter('collector_address', collector_address),
+        ...creatEqFilter('chain', chain),
+        ...creatEqFilter('drop_id', drop_id),
+        ...createBetweenFilter('minted_on', minted_date_from, minted_date_to),
+        ...createInFilter('id', ids),
+      },
+    };
+
+    const { data } = await this.CompassProvider.request<PaginatedPoapsResponse>(
+      PAGINATED_POAPS_QUERY,
+      variables,
+    );
+
+    const poaps = data.poaps.map((poap) => {
+      const { drop } = poap;
+      return new POAP({
+        ...poap,
+        ...drop,
+        id: Number(poap.id),
+        minted_on: new Date(poap.minted_on),
```

Comment:
What the difference between `minted_on` and `created_date`?

Isn't `expiry_date` missing here?

Reply:
The expiry date is more information about the drop not the POAP itself i think

Reply:
Right.